soco/events_base.py

```diff
@@ -0,0 +1,642 @@
+        # CALLBACK: <delivery URL>
+        # NT: upnp:event
+        # TIMEOUT: Second-requested subscription duration (optional)
+
+        # pylint: disable=unbalanced-tuple-unpacking
+        ip_address, port = self.event_listener.address
+        headers = {
+            'Callback': '<http://{}:{}>'.format(ip_address, port),
+            'NT': 'upnp:event'
+        }
+        if requested_timeout is not None:
+            headers["TIMEOUT"] = "Second-{}".format(requested_timeout)
+
+        # pylint: disable=missing-docstring
+        def success(headers):
+            self.sid = headers['sid']
+            timeout = headers['timeout']
+            # According to the spec, timeout can be "infinite" or "second-123"
+            # where 123 is a number of seconds.  Sonos uses "Second-123"
+            # (with a capital letter)
+            if timeout.lower() == 'infinite':
+                self.timeout = None
+            else:
+                self.timeout = int(timeout.lstrip('Second-'))
+            self._timestamp = time.time()
+            self.is_subscribed = True
+            log.info(
+                "Subscribed to %s, sid: %s",
+                service.base_url + service.event_subscription_url, self.sid)
+            # Register the subscription so it can be looked up by sid
+            # and unsubscribed at exit
+            self.subscriptions.register(self)
+
+            # Set up auto_renew
+            if not auto_renew:
+                return
+            # Autorenew just before expiry, say at 85% of self.timeout seconds
+            interval = self.timeout * 85 / 100
+            self.auto_renew_start(interval)
+
+        # pylint: disable=missing-docstring
+        def failure():
+            raise SoCoException(
+                "Could not subscribe to {}".format(
+                    service.base_url + service.event_subscription_url))
+
+        # Lock out EventNotifyHandler during registration.
+        # If events_twisted is used, this lock should always be
+        # available, since threading is not being used.
+        with self.subscriptions.subscriptions_lock:
+            self.request(
+                'SUBSCRIBE', service.base_url + service.event_subscription_url,
+                headers, success, failure)
+
+    def renew(self, requested_timeout=None):
+        """renew(requested_timeout=None)
+        Renew the event subscription.
+        You should not try to renew a subscription which has been
+        unsubscribed, or once it has expired.
+
+        Args:
+            requested_timeout (int, optional): The period for which a renewal
+                request should be made. If None (the default), use the timeout
+                requested on subscription.
+
+        """
+        # NB This code may be called from a separate thread when
+        # subscriptions are auto-renewed. Be careful to ensure thread-safety
+
+        log.info("Autorenewing subscription %s", self.sid)
```

Comment:
The message '**Auto**renewing subscription %s' might not always be accurate. Just make it 'Renewing subscription %s'.